src/test/groovy/com/calgaryscientific/gradle/VeracodeTaskTest.groovy

```diff
@@ -192,16 +194,69 @@ class VeracodeTaskTest extends Specification {
 
         when:
         project.plugins.apply('com.calgaryscientific.gradle.veracode')
-        project.ext.veracodeSetup = vs
+        project.veracodeSetup.filesToUpload = vs.filesToUpload
 
         then:
         VeracodeSetup vsRead = project.findProperty("veracodeSetup") as VeracodeSetup
         Set<File> expected = [buildFile] as Set
         assert vsRead.filesToUpload == expected
-        def task = project.tasks.getByName("veracodeUploadFile")
+        VeracodeUploadFileTask task = project.tasks.getByName("veracodeUploadFile") as VeracodeUploadFileTask
+        assert task.veracodeSetup.filesToUpload == vsRead.filesToUpload
         assert task.getFileSet() == expected
         def _ = vsRead.filesToUpload.add(buildFile)
         assert vsRead.filesToUpload == [buildFile, buildFile] as Set
     }
 
+    def 'Test VeracodeUploadFile Task'() {
+        given:
+        VeracodeAPI veracodeAPIMock = Mock(VeracodeAPI, constructorArgs: ["", "", null, null])
+        String xmlStr = '''<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<filelist xmlns="something" xmlns:xsi="something" filelist_version="1.1">
+    <file file_id="1" file_md5="d98b6f5ccfce3799e9b60b5d78cc1" file_name="file1" file_status="Uploaded"/>
+    <file file_id="2" file_md5="68a7d8468ca51bc46d5b72d485022" file_name="file2" file_status="Uploaded"/>
+    <file file_id="3" file_md5="2459464ff4bf78dd6f09695069b52" file_name="file3" file_status="Uploaded"/>
+</filelist>
+'''
+        Project project = new ProjectBuilder().build()
+        project.plugins.apply('com.calgaryscientific.gradle.veracode')
+        VeracodeSetup vs = new VeracodeSetup()
```

Comment:
ah .... VeracodeSetup it contains list of files to be uploaded ...., should we rename that to something like "artifactsTobeSubmiittedToVeracode", what's the domain term the veracode team use to indicate the artifacts to be submitted?

Reply:
See comment above, the Veracode actual API just as a method call called `uploadFile`. Since the VeracodeSetup is the only thing you have to setup for our plugin, the field name doesn't seem to be any more specific than `VeracodeSetup.filesToUpload`, see usage in the readme here:

https://github.com/calgaryscientific/veracode-gradle-plugin#apply-the-plugin-into-your-project